### PR TITLE
Capture overlap bits from TrueType glyphs

### PR DESCRIPTION
**Pull request by dfrg**
First step in addressing #581. Captures simple and composite overlap bits and plumbs them through internally.

Still needs to be exposed in skrifa public API. This will be done in a follow up.

### CODE REVIEW

File: read-fonts/src/tables/glyf.rs [hunk 1]
```diff
@@ -945,6 +958,52 @@ mod tests {
...
+            let glyph = match loca.get_glyf(GlyphId::new(gid), &glyf) {
+                Ok(Some(Glyph::Simple(glyph))) => glyph,
+                _ => continue,
+            };
+            if gid == 3 {
+                // Only GID 3 has the overlap bit set
+                assert!(glyph.has_overlapping_contours())
+            } else {
+                assert!(!glyph.has_overlapping_contours())
+            }
```
Comment (rsheeter) [inline read-fonts/src/tables/glyf.rs:977]: This seems like the verbose version of  `assert_eq!(gid == 3, glyph.has_overlapping_contours())` ?

Comment (rsheeter) [reply read-fonts/src/tables/glyf.rs:977]: Hm. Actually even then it's a bit more vulnerable to broken test data than I like; I propose we directly assert that only gid 3 has the bit:

`assert_eq!(vec![3], (0..glyph_count).filter(...retain only if has overlaps).collect())`

File: read-fonts/src/tables/glyf.rs [hunk 2]
```diff
@@ -945,6 +958,52 @@ mod tests {
...
+        }
+    }
+
+    #[test]
+    fn composite_overlapping_contour_flag() {
```
Comment (rsheeter) [inline read-fonts/src/tables/glyf.rs:982]: The loading of test data, num glyphs, etc is repetitive noise; perhaps a test helper that lets you easily iterate over (gid, &Glyph) would help tidy?

File: read-fonts/src/tables/glyf.rs [hunk 3]
```diff
@@ -945,6 +958,52 @@ mod tests {
...
+        let font = FontRef::new(font_test_data::VAZIRMATN_VAR).unwrap();
+        let loca = font.loca(None).unwrap();
+        let glyf = font.glyf().unwrap();
+        let glyph_count = font.maxp().unwrap().num_glyphs();
+        for gid in 0..glyph_count {
+            let glyph = match loca.get_glyf(GlyphId::new(gid), &glyf) {
+                Ok(Some(Glyph::Composite(glyph))) => glyph,
+                _ => continue,
+            };
+            // Only GID 2, component 1 has the overlap bit set
```
Comment (rsheeter) [inline read-fonts/src/tables/glyf.rs:992]: As above I might prefer it as something like `assert_eq!(vec![(2, 1)], ...collect (gid, component index) of things with overlap compound)`

File: skrifa/src/scale/glyf/scaler.rs [hunk 4]
```diff
@@ -733,3 +735,22 @@ impl<'a, H> ScalerInstance<'a, H> {
...
+    fn overlap_flags() {
+        let font = FontRef::new(font_test_data::VAZIRMATN_VAR).unwrap();
+        let scaler = Scaler::new(&font).unwrap();
+        let glyph_count = font.maxp().unwrap().num_glyphs();
+        for gid in 0..glyph_count {
+            let glyph = scaler.glyph(GlyphId::new(gid), false).unwrap();
+            // GID 2 is a composite glyph with the overlap bit on a component
+            // GID 3 is a simple glyph with the overlap bit on the first flag
+            assert_eq!(glyph.has_overlaps, gid == 2 || gid == 3);
+        }
```
Comment (rsheeter) [inline skrifa/src/scale/glyf/scaler.rs:754]: As above, I'd prefer `assert_eq!(vec[2, 3], ...collect gids with overlap...)`

**Review by rsheeter — APPROVED** [review]
LGTM

File: read-fonts/src/tables/glyf.rs (hunk 1 above)
Comment (dfrg) [reply read-fonts/src/tables/glyf.rs:977]: I like this style as well. Fixed!

File: read-fonts/src/tables/glyf.rs (hunk 2 above)
Comment (dfrg) [reply read-fonts/src/tables/glyf.rs:982]: The refactoring part of my brain seems to turn off when writing tests. Good call, ty!

**Comment by dfrg** [comment]
Oops, I pushed all the fixes requested here to the branch for #583. Since both PRs are approved and related, I'll consider these "fixed in a followon" :)

**Pull request merged**